### PR TITLE
Adds encrypted flag to round metadata, only email is set to true for now

### LINKED ISSUE #221: Add encryption boolean to round metadata
**User Story:**
As a RM engineer
I want to pass encryption information in my application schema
So that Grant Hub knows which fields to encrypt

**Acceptance Criteria**
- ~Public key can be deployed with round schema~
- ~Encryption details are added to round application schema~
- Make it possible to add encrypt as a boolean to every question. default it to false
- Ensure when the application schema is updated , the documentation is updated as well 
- For MVP, email is the only field which should be encrypted 

**Product & Design Links:**

**Tech Details:**

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by LukeDowell**
Resolves #221 

Round contract: 
0x001b73d0c92f50e2a5385b1328d615d2addac438

Application Metadata:
https://gitcoin.mypinata.cloud/ipfs/bafkreifkt4sond4c5amadjffnoy6cebxbmflcid5trmxazf5os3zlpwi2a

### CODE REVIEW

File: packages/round-manager/src/features/api/utils.ts [hunk 1]
```diff
@@ -176,37 +176,35 @@ export const abbreviateAddress = (address: string) => `${address.slice(0, 8)}...
...
-      schema.push({
-        id: schema.length,
-        question: camelToTitle(key),
-        type: "TEXT",
-        required: true,
-        info: metadata[key],
-        choices: []
-      })
-    }
+export const generateApplicationSchema = (metadata: ApplicationMetadata) => {
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/api/utils.ts:179]: Could we add the return type as an array ? Easier for doc generation

File: packages/round-manager/src/features/api/utils.ts [hunk 2]
```diff
@@ -176,37 +176,35 @@ export const abbreviateAddress = (address: string) => `${address.slice(0, 8)}...
...
+  if (!metadata.customQuestions) return []
+
+  const maybe_questions = metadata.customQuestions
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/api/utils.ts:182]:
```suggestion
  const customQuestions = metadata.customQuestions
```

let's keep the coding style consistent and keep it camelCase
Additionally let's avoid renaming variables. might cause confusion down the line

File: packages/round-manager/src/features/api/utils.ts [hunk 3]
```diff
@@ -176,37 +176,35 @@ export const abbreviateAddress = (address: string) => `${address.slice(0, 8)}...
...
+
+  return questions.map(([name , answer], i) => {
+    return {
+      id: i,
+      question: camelToTitle(name),
+      type: "TEXT",
+      required: true,
+      info: answer,
+      choices: [],
+      encrypted: name === 'email'
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/api/utils.ts:207]: worth adding a TODO comment that we would need to make remove this hardcoding and make it dynamic ?

**Review by thelostone-mc — DISMISSED** [review]
@LukeDowell pointing to this https://discord.com/channels/562828676480237578/973967228577935400/1006868185116585984

Could we mark 
- which issue this fixes
- share the round contract where this applcation form was created (with the encrypted email)
we would need this info to share it to the grant-hub team as well

^ Ideally throw that in the PR description adding a bit more context

File: packages/round-manager/src/features/api/utils.ts (hunk 2 above)
Comment (LukeDowell) [reply packages/round-manager/src/features/api/utils.ts:182]: Ha you are totally right, I have been doing python recently and it slipped in

File: packages/round-manager/src/features/api/utils.ts (hunk 3 above)
Comment (LukeDowell) [reply packages/round-manager/src/features/api/utils.ts:207]: Personally, I don't much mind the static nature of the type checking here. I guess in this case it's all strings and wild-westy, but my preference would be to take the hit of having to update this when the input changes as a way of keeping ourselves honest about what we add to the metadata. 

Either way I'll add the TODO

**Review by chibie — CHANGES_REQUESTED** [review]
@LukeDowell the round application schema doc will have to be updated to include the `encrypt` field

`packages/contracts/docs/ROUND_APPLICATION.md#L26`

**Review by shavinac — APPROVED** [review]

**Pull request merged**